yente/search/indexer.py

```diff
@@ -32,27 +33,38 @@
     NAMES_FIELD,
     NAME_PHONETIC_FIELD,
 )
+from yente.search.util import parse_index_name
 from yente.search.util import construct_index_name, construct_index_version
 from yente.data.util import expand_dates, phonetic_names
 from yente.data.util import index_name_parts, index_name_keys
 
+
 log = get_logger(__name__)
 
 
 async def iter_entity_docs(
-    dataset: Dataset, index: str
+    loader: DatasetLoader, index: str
 ) -> AsyncGenerator[Dict[str, Any], None]:
-    if dataset.entities_url is None:
-        return
+    dataset = loader.dataset
     datasets = set(dataset.dataset_names)
     idx = 0
-    async for data in load_json_lines(dataset.entities_url, index):
+    ops: Dict[str, int] = {"ADD": 0, "DEL": 0, "MOD": 0}
+    async for data in loader.load():
         if idx % 1000 == 0 and idx > 0:
             log.info("Index: %d entities..." % idx, index=index)
+        op_code = data["op"]
         idx += 1
+        ops[op_code] += 1
```

Comment:
Here I think it's better to keep the construction of the EntityOp in the SearchProvider or this will become a mess the day OpenSearch and ElasticSearch diverge.

Reply:
I feel like the EntityOp is sort of the interface between the loader and the provider - then turning it into ES/OS ops is where it becomes engine-specific?